modules/javafx.graphics/src/main/java/javafx/scene/text/Text.java

```diff
@@ -1021,27 +1021,35 @@ public final BooleanProperty caretBiasProperty() {
     public final HitInfo hitTest(Point2D point) {
         if (point == null) return null;
         TextLayout layout = getTextLayout();
+
         double x = point.getX() - getX();
         double y = point.getY() - getY() + getYRendering();
         GlyphList[] runs = getRuns();
-        int runIndex = 0;
-        if (runs.length != 0) {
-            double ptY = localToParent(x, y).getY();
-            while (runIndex < runs.length - 1) {
-                if (ptY > runs[runIndex].getLocation().y && ptY < runs[runIndex + 1].getLocation().y) {
-                    break;
-                }
-                runIndex++;
-            }
-        }
         int textRunStart = 0;
-        int curRunStart = 0;
         if (runs.length != 0) {
-            textRunStart = ((TextRun) runs[0]).getStart();
-            curRunStart = ((TextRun) runs[runIndex]).getStart();
+            textRunStart = findFirstRunStart(runs);
+        }
```

Comment:
Updated `findFirstRunStart` method. I believe the length test is not really necessary here because even if the Text node is empty, a run will be created for it. So did not add length check.